samples/client/petstore/r_test/R/PetApi.r

```diff
@@ -0,0 +1,161 @@
+# Swagger Petstore
+# 
+# This is a sample server Petstore server.  You can find out more about Swagger at [http://swagger.io](http://swagger.io) or on [irc.freenode.net, #swagger](http://swagger.io/irc/).  For this sample, you can use the api key `special-key` to test the authorization filters.
+# 
+# OpenAPI spec version: 1.0.0
+# Contact: apiteam@swagger.io
+# Generated by: https://github.com/swagger-api/swagger-codegen.git
+
+PetApi <- R6::R6Class(
+  'PetApi',
+  public = list(
+    userAgent = "Swagger-Codegen/1.0.0/r",
+    host = NULL,
+    basePath = NULL,
+    scheme = NULL,
+    url = NULL,
+    initialize = function(host, basePath, scheme){
```

Comment:
Can `host`, `basePath`, `scheme` be optional and we'll provide the default value based on the spec? For example, the default value for basePath is `v2`: https://github.com/swagger-api/swagger-codegen/blob/master/modules/swagger-codegen/src/test/resources/2_0/petstore.yaml#L13

Reply:
default values can be set during definition itself. To allow for optional arguments in initialize function, we just need to check whether the argument is "missing". Here is an example of setting default values and optional arguments in PetApi class : 

PetApi <- R6::R6Class(
  'PetApi',
  public = list(
    userAgent = "Swagger-Codegen/1.0.0/r",
    host = "petstore.swagger.io",
    basePath = "v2",
    scheme = "http",
    url = NULL,
    initialize = function(host, basePath, scheme){
      if (!missing(host)) {
        stopifnot(is.character(host), length(host) == 1)
        self$host <- host
      }
      if (!missing(basePath)) {
        stopifnot(is.character(basePath), length(basePath) == 1)
        self$basePath <- basePath
      }
      if (!missing(scheme)) {
        stopifnot(is.character(scheme), length(scheme) == 1)
        self$scheme <- scheme
      }
      self$url <- sprintf("%s://%s/%s/pet/", scheme, host, basePath)
    }

Reply:
Nice. Will incorporate these changes later today.